workers/data_refinery_workers/processors/janitor.py

```diff
@@ -59,21 +59,32 @@ def _find_and_remove_expired_jobs(job_context):
                     # This job is running, don't delete  the working directory.
                     if job_status == "running":
                         continue
+                except URLNotFoundNomadException as e:
+                    # If we can't currently access Nomad,
+                    # just continue until we can again.
+                    continue
                 except BaseNomadException as e:
                     # If we can't currently access Nomad,
                     # just continue until we can again.
                     continue
                 except Exception as e:
                     # This job is likely vanished. No need for this directory.
-                    pass
+                    # Or, possibly, another Nomad error outside of BaseNomadException.
+                    logger.exception("Janitor found vanished job for " + item + " - why?")
+                    continue
+            except ProcessorJob.DoesNotExist:
+                # This job has vanished from the DB - clean it up!
+                logger.exception("Janitor found no record of " + item + " - why?")
+                pass
             except Exception:
                 # If we don't have a record of the job, we don't need the directory.
-                pass
-
+                logger.exception("Problem finding job record for " + item + " - why?")
+                continue
```

Comment:
The comment above this now seems out of date.

Reply:
There are two separate cases here, that we are handling in different ways. The first (object DNE) - then we say we found no record of it, and we `pass` so it gets deleted. If there is another type of Exception, that means that we are likely currently unable to access the database (PgBouncer is down, for example), so we don't want to perform any destructive action yet, hence the continue and "problem finding record".

Reply:
Right, but I'm talking about the `# If we don't have a record of the job, we don't need the directory.` comment which used to precede the `pass`, but now it's before a continue so I don't think it makes sense.